extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/aggregation/bloom/BloomFilterAggregatorFactory.java

```diff
@@ -0,0 +1,233 @@
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory columnFactory)
+  {
+    ColumnSelectorPlus<BloomFilterAggregatorColumnSelectorStrategy> selectorPlus =
+        DimensionHandlerUtils.createColumnSelectorPlus(
+            STRATEGY_FACTORY,
+            field,
+            columnFactory
+        );
+
+    return new BloomFilterBufferAggregator(selectorPlus, maxNumEntries);
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    // idk how to compare?
+    return (Comparator<BloomKFilter>) (o1, o2) -> 0;
+  }
+
+  @Override
+  public Object combine(Object lhs, Object rhs)
+  {
+    if (rhs == null) {
+      return lhs;
+    }
+    if (lhs == null) {
+      return rhs;
+    }
+    ((BloomKFilter) lhs).merge((BloomKFilter) rhs);
+    return lhs;
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new BloomFilterMergeAggregatorFactory(name, name, maxNumEntries);
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.singletonList(new BloomFilterAggregatorFactory(name, field, maxNumEntries));
+  }
+
+  @Override
+  public Object deserialize(Object object)
+  {
+    final ByteBuffer buffer;
+
+    if (object instanceof byte[]) {
+      buffer = ByteBuffer.wrap((byte[]) object);
+    } else if (object instanceof ByteBuffer) {
+      // Be conservative, don't assume we own this buffer.
+      buffer = ((ByteBuffer) object).duplicate();
+    } else if (object instanceof String) {
+      buffer = ByteBuffer.wrap(Base64.decodeBase64(StringUtils.toUtf8((String) object)));
+    } else {
+      return object;
+    }
+
+    ByteBufferInputStream byteBufferInputStream = new ByteBufferInputStream(buffer);
+    try {
+      return BloomKFilter.deserialize(byteBufferInputStream);
+    }
+    catch (Exception ex) {
+      throw new RuntimeException("Failed to deserialize bloomK filter", ex);
+    }
+  }
+
+  @Override
+  public Object finalizeComputation(Object object)
+  {
+    return object;
+  }
+
+  @JsonProperty
+  @Override
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public DimensionSpec getField()
+  {
+    return field;
+  }
+
+  @JsonProperty
+  public int getMaxNumEntries()
+  {
+    return maxNumEntries;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(field.getDimension());
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return "bloomFilter";
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    BloomKFilter throwaway = new BloomKFilter(maxNumEntries);
+    return (throwaway.getBitSet().length * Long.BYTES) + 5;
```

Comment:
What is +5?

Reply:
It's a header written during serialization I'll add a comment/link to https://github.com/apache/hive/blob/master/storage-api/src/java/org/apache/hive/common/util/BloomKFilter.java#L302 or see if I can find a better way to get this information

Reply:
Now uses new method on copied `BloomKFilter` to compute the size